### PR TITLE
End to End tests for local authentication.

### PR DESCRIPTION
**Pull request by andela-jmuturi**
#### What does this PR do?
Adds end to end local authentication tests.

#### Description of Task to be completed?
Write e2e tests using Protractor and Selenium to test the local authentication workflow.

#### How should this be manually tested?
Clone this repo, install the dependencies by running
```
npm install
```
Make sure you have Java (JDK) installed in your system then run:
```
npm run e2esetup
```
This will spin up the Selenium webserver and install Selenium weddriver if it doesn't exist in your system.
Then run:
```
gulp e2etest
```

#### What are the relevant pivotal tracker stories?
[E2E Test Configuration](https://www.pivotaltracker.com/projects/1454736/stories/117129339)
[E2E Test: For authentication](https://www.pivotaltracker.com/projects/1454736/stories/117129333)

@hisabimbola @andela-ekupara @andela-kndungu

### CODE REVIEW

File: tests/e2e/home-test.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/* global browser, by*/
+(function() {
+  'use strict';
+
+  describe('Trime E2E Test Suite:', function() {
+    var homePage = 'http://localhost:3000';
```
Comment (andela-jmuturi) [inline tests/e2e/home-test.js]: Move this to protractor's baseurl setting.

File: tests/e2e/login-signup-page-test.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    var driver = browser.driver;
+    var authPage = homePage + '/join';
+    var credentials = require('./helpers/helpers').credentials;
+
+    describe('Trime signup/login page:', function() {
+      beforeEach(function() {
+        browser.get(authPage);
+        var width = 1366;
+        var height = 768;
+        browser.driver.manage().window().setSize(width, height);
```
Comment (andela-jmuturi) [inline tests/e2e/login-signup-page-test.js]: Move the height and width config to protractor config.

File: tests/e2e/login-signup-page-test.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+      it('should contain local login button', function() {
+        var button = driver.findElement(
+          by.css('form > div.col.s12.center-align button'));
+        expect(button.getText()).toEqual('LOGIN');
+      });
+    });
+
+    describe('User authentication:', function() {
+      var newUser = {
+        email: 'test' + Math.floor(Math.random() * 1000000) + '@test.com'
```
Comment (andela-jmuturi) [inline tests/e2e/login-signup-page-test.js]: Use Date.now instead of Math.random

File: tests/e2e/login-signup-page-test.js (hunk 2 above)
Comment (andela-ekupara) [inline tests/e2e/login-signup-page-test.js]: The handling of the browser size is handled by the Selenium driver and it wouldn't make sense for it to go into protractor configurations. Using the driver management methods is the only way I have seen a number of people handle it across several stacks.

**Pull request merged**